### PR TITLE
Feature/iteration1 script refactor

### PR DESCRIPTION
**Pull request by Shakikka**
## What does this PR do?

- Refactors scripts.js to incorporate Fetch API rather than data files for user information.
- Add average daily steps to UI

## How should it be tested in the UI?

- Open the website and look at the left hand column.

### CODE REVIEW

**Comment by Shakikka** [comment]
[Refactor scripts.js *Iteration 1*](https://trello.com/c/2uJQPk8h/30-refactor-scriptsjs)

**Comment by Shakikka** [comment]
[DOM: Step goal comparison vs all user average not displayed](https://trello.com/c/JPui63eD/20-dom-step-goal-comparison-vs-all-user-average-not-displayed)

File: src/User.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ class User {
     return this.name.split(' ', 1).join();
   }
   getFriendsNames(userStorage) {
-    return this.friends.map((friendId) => (userStorage.getDataFromID(friendId).name));
+    return this.friends.map((friendId) => (userStorage.findUserData(friendId).name));
```
Comment (m1073496) [inline src/User.js:16]: Nice catch

File: src/scripts.js [hunk 2]
```diff
@@ -18,86 +19,132 @@ import {
...
+const avgStairsToday = document.getElementById('avgStairsToday');
+const userMinutesToday = document.getElementById('userMinutesToday');
+const avgMinutesToday = document.getElementById('avgMinutesToday');
+const userStepsThisWeek = document.getElementById('userStepsThisWeek');
+const userStairsThisWeek = document.getElementById('userStairsThisWeek');
+const userMinutesThisWeek = document.getElementById('userMinutesThisWeek');
+const bestUserSteps = document.getElementById('bestUserSteps');
+const streakList = document.getElementById('streakList');
+const streakListMinutes = document.getElementById('streakListMinutes')
+const avStepGoalCard = document.getElementById('avStepGoalCard');
```
Comment (m1073496) [inline src/scripts.js:54]: Thanks for changing these!

File: src/scripts.js [hunk 3]
```diff
@@ -18,86 +19,132 @@ import {
...
+// .then(data => console.log('sleep:', data.sleepData))
+// userDataAPI
+// .then(data => console.log('user:', data.userData))
+// activityDataAPI
+// .then(data => console.log('activity:', data.activityData))
+// hydrationDataAPI
+// .then(data => console.log('hydration:', data.hydrationData))
+
+function loadThisFirst() {
+  Promise.all([sleepDataAPI, userDataAPI, activityDataAPI, hydrationDataAPI])
```
Comment (m1073496) [inline src/scripts.js:65]: Great use of Promise.all --> I'd love to discuss this tomorrow

File: src/scripts.js [hunk 4]
```diff
@@ -18,86 +19,132 @@ import {
...
+    let today = makeToday(userRepo, userNowId, hydrationData);
+    let randomHistory = makeRandomDate(userRepo, userNowId, hydrationData);
+    historicalWeek.forEach(instance => instance.insertAdjacentHTML('afterBegin', `Week of ${randomHistory}`));
+    // addInfoToSidebar(userNow, userRepo);
+    addHydrationInfo(userNowId, hydrationRepo, today, userRepo, randomHistory);
+    addSleepInfo(userNowId, sleepRepo, today, userRepo, randomHistory);
+    let winnerNow = makeWinnerID(activityRepo, userNow, today, userRepo);
+    addActivityInfo(userNowId, activityRepo, today, userRepo, randomHistory, userNow, winnerNow);
+    addFriendGameInfo(userNowId, activityRepo, userRepo, today, randomHistory, userNow);
+  })
```
Comment (m1073496) [inline src/scripts.js:88]: So this stuff now fires only after the promises are resolved?

File: src/scripts.js [hunk 5]
```diff
@@ -185,4 +232,5 @@ function makeStepStreakHTML(id, activityInfo, userStorage, method) {
   return method.map(streakData => `<li class="historical-list-listItem">${streakData}!</li>`).join('');
 }
 
-startApp();
\ No newline at end of file
+// startApp();
+window.addEventListener('load', loadThisFirst)
```
Comment (m1073496) [inline src/scripts.js:236]: Sweet!

File: src/scripts.js [hunk 6]
```diff
@@ -18,86 +19,132 @@ import {
...
+//   addHydrationInfo(userNowId, hydrationRepo, today, userRepo, randomHistory);
+//   addSleepInfo(userNowId, sleepRepo, today, userRepo, randomHistory);
+//   let winnerNow = makeWinnerID(activityRepo, userNow, today, userRepo);
+//   addActivityInfo(userNowId, activityRepo, today, userRepo, randomHistory, userNow, winnerNow);
+//   addFriendGameInfo(userNowId, activityRepo, userRepo, today, randomHistory, userNow);
+// }
+
+
+
+
```
Comment (m1073496) [inline src/scripts.js:147]: Glad to see this gone, nice work!

**Review by m1073496 — APPROVED** [review]

**Pull request merged**